qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -100,6 +126,50 @@ def default_values(self) -> List[Tuple[ParameterValueType, Parameter, Tuple, Sch
             :class:`Calibrations` can call :meth:`add_parameter_value` on the tuples.
         """
 
+    @abstractmethod
+    def _build_schedules(self):
+        """Build the schedules stored in the library.
+
+        This method is called as the last step in the :meth:`__init__`. Subclasses must implement
+        :meth:`_build_schedules` to build the schedules of the library based on the inputs given
+        to the :meth:`__init__` method.
+        """
+
+    @property
+    def config(self) -> Dict[str, Any]:
+        """Return the settings used to initialize the library."""
+
+        kwargs = {"basis_gates": self.basis_gates, "default_values": self._default_values}
+        kwargs.update(self._extra_kwargs)
+
+        return {
+            "class": self.__class__.__name__,
+            "kwargs": kwargs,
+            "hash": self.__hash__(),
```

Comment:
The hash is a consequence of the configuration. The hash validation is a solution to the library versioning problem. We want to ensure that a deserialized library matches the serialized one. If the code of the library changes in between serialization and deserialization (e.g. because we are actively developing it) you will have a problem. This problem is detected by the hash.

Reply:
Yes, this is somewhat a consequence of not being able to serialize schedules, so we need to rely on the same inputs to the library producing the same schedules as when the library was serialized. However, when we have schedule serialization, we will need to decide whether to rely on the previously serialized schedules or to check if the library now produces different schedules.

Reply:
I understand but such check doesn't exist in experiment framework. Experiment encoder stores the system version but only raises the error when exception happens (so no active validation like this).

Reply:
This is so specific to the library that I would avoid building this mechanism into the experiment framework. Furthermore, this is also temporary and will not be needed once we can serialize the schedules. I did however add a test for the behavior we want to capture. See here: b7a79b3

Reply:
I think this is an issue we should discuss further outside of this PR -- when should something be serialized as a data container vs. a class (should the data itself be serialized or the init args) and what validation should be done on deserialization.

Naoki's point might be that `config` here is violating expectations based on previous usage in `ExperimentConfig`. A possible solution might be to do the validation / hashing in the `__json_decode__` and `__json_encode__` methods only. I think those have less assumptions built in than `config` does.